### PR TITLE
rename "microbit" package to "core"

### PR DESCRIPTION
**Pull request by microsoftsam**
- add a packageMap to the compile configuration to remap package (microbit -> core)
- remove old mapping feature for microbit-(led|game...) and use config instead

### CODE REVIEW

**Comment by microsoftsam** [comment]
Microsoft/pxt-microbit#270

**Comment by riknoll** [comment]
It would be nice to add some tests for importing hex files. I'm guessing that most of that code is pretty reliant on running in a browser at the moment?

File: pxtrunner/runner.ts [hunk 1]
```diff
@@ -336,7 +336,7 @@ ${files["main.ts"]}
 
     function renderDocAsync(content: HTMLElement, docid: string): Promise<void> {
         docid = docid.replace(/^\//, "");
-        return pxt.Cloud.privateGetTextAsync(`md/${pxt.appTarget.id}/${docid}`)
+        return pxt.Cloud.privateGetTextAsync(`md/${pxt.appTarget.nickname}/${docid}`)
```
Comment (mmoskal) [inline pxtrunner/runner.ts]: What is this?

File: webapp/public/custom.css [hunk 2]
```diff
@@ -296,6 +296,16 @@ div.simframe > iframe {
...
+.blocklyTreeRow {
+    padding-top: 0.75rem !important;
+    padding-bottom: 2rem !important;
+    border-left-width: 12px !important;
+}
+
+.blocklyTreeSelected {
+    border-radius: 5px !important;
+}
+
```
Comment (mmoskal) [inline webapp/public/custom.css]: Doesn't seem related to package upgrade

File: webapp/src/package.ts [hunk 3]
```diff
@@ -278,9 +278,9 @@ class Host
     }
 
     writeFile(module: pxt.Package, filename: string, contents: string): void {
-        if (filename == pxt.CONFIG_NAME)
-            return; // ignore config writes
-        throw Util.oops("trying to write " + module + " / " + filename)
+        let epkg = getEditorPkg(module)
+        let file = epkg.files[filename];
+        file.content = contents;
```
Comment (mmoskal) [inline webapp/src/package.ts]: I would limit this to config file updates. This will generally not flush the file to "disk" (local storage in the browser or disk via localhost).

File: pxtrunner/runner.ts (hunk 1 above)
Comment (mmoskal) [inline pxtrunner/runner.ts]: I think this is wrong. It should be 'id'. Cloud doesn't know about nicknames

File: webapp/public/custom.css (hunk 2 above)
Comment (microsoftsam) [inline webapp/public/custom.css]: this's already been checked into master.

**Comment by pelikhan** [comment]
nickname meant for file naming, human readable version of id.

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

File: pxtlib/main.ts [hunk 4]
```diff
@@ -253,12 +253,41 @@ namespace pxt {
...
+        }
+
         private parseConfig(str: string) {
             let cfg = <PackageConfig>JSON.parse(str)
             this.config = cfg;
-            // temp patch for cloud corrupted configs
-            for (let dep in this.config.dependencies)
-                if (/^microbit-(led|music|game|pins|serial)$/.test(dep)) delete this.config.dependencies[dep];
+
+            let configChanged = false;
```
Comment (pelikhan) [inline pxtlib/main.ts]: you could serialize config before and after to guarantee that nothing changed

File: webapp/src/package.ts [hunk 5]
```diff
@@ -278,9 +278,12 @@ class Host
     }
 
     writeFile(module: pxt.Package, filename: string, contents: string): void {
-        if (filename == pxt.CONFIG_NAME)
-            return; // ignore config writes
-        throw Util.oops("trying to write " + module + " / " + filename)
```
Comment (pelikhan) [inline webapp/src/package.ts]: no more "oops"?
